### DISCUSSION TITLE
Set the block or boundary IDs for the mesh in MOOSE

### DISCUSSION THREAD

**Discussion by wenpeng231024** [idaholab/moose#16233] [Q&A General] [answered]
Hi everyone,

I want to set the block or boundary IDs for my mesh in MOOSE. I use the "GeneratedMesh" type for mesh. Does anyone know how to do this?

Best,

Peng

**Comment by aeslaughter** [comment, marked as answer]
There are a few methods of doing this, one of the most common is to use a SubdomainBoundingBoxGenerator to assign an id within a region.

https://mooseframework.inl.gov/source/meshgenerators/SubdomainBoundingBoxGenerator.html

Look at the various input files listed at the bottom of the page for examples.

**Reply by wenpeng231024** [reply]
Hi Andrew,

Thanks for your help. I will try this.

Peng
